network/discovery/dv5_service.go

```diff
@@ -124,15 +128,33 @@ func (dvs *DiscV5Service) Bootstrap(logger *zap.Logger, handler HandleNewPeer) e
 	zeroSubnets, _ := records.Subnets{}.FromString(records.ZeroSubnets)
 
 	dvs.discover(dvs.ctx, func(e PeerEvent) {
+		logger := logger.With(
+			fields.ENR(e.Node),
+			fields.PeerID(e.AddrInfo.ID),
+		)
+
+		// Get the peer's domain type, skipping if it mismatches ours.
+		nodeDomainType, err := records.GetDomainTypeEntry(e.Node.Record())
+		if err != nil {
+			logger.Debug("could not read domain type", zap.Error(err))
+			return
+		}
+		if nodeDomainType != dvs.domainType {
+			logger.Debug("skipping node with different domain type")
+			return
+		}
+
```

Comment:
@moshe-blox 
Due to it being not backward compatible and requiring a fork, let's not enforce the check.
and add it once we ready with fork support

Reply:
done

Reply:
https://github.com/bloxapp/ssv/pull/1039/files#diff-f608add482b1df6a8efdebfbc895809e63ffca366b76ffd8192666b0f132ea9cR21